### PR TITLE
Bisect for Binary Search

### PR DESCRIPTION
**Pull request by muhahahmad68**
## Overview: What does this pull request change?

Implemented the binary search using the python bisect module

## Motivation and Explanation: Why and how do your changes improve the library?

Makes the function more precise
Solving issue #3437 
## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/utils/simple_functions.py [hunk 1]
```diff
@@ -54,28 +55,16 @@ def binary_search(
...
-            else:
-                lh = mh
-        elif lx > target > rx:
-            lh, rh = rh, lh
-        else:
-            return None
-
-    return mh
+    inputs = [
+        lower_bound + i * (upper_bound - lower_bound) / 10000 for i in range(10001)
```
Comment (behackl) [inline manim/utils/simple_functions.py:59]: 10000 seems like a rather arbitrary value. Where does it come from?

Comment (muhahahmad68) [reply manim/utils/simple_functions.py]: Used in this code to generate a list of 10,000 evenly spaced values between lower_bound and upper_bound with list comprehension

Comment (muhahahmad68) [reply manim/utils/simple_functions.py]: Arbitrary fr
I'm surfing the net to find ways to implement the function better using the bisect module

File: manim/utils/simple_functions.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 ]
 
 
+import bisect
```
[the remark this answers is not in the document]
Comment (muhahahmad68) [reply manim/utils/simple_functions.py:13]: I'll rectify that

**Comment by behackl** [comment]
See the latest comment in #3437, after more careful consideration I don't see a straightforward way to replace our bisection by Python's standard `bisect` function. I am closing this PR.

**Pull request closed without merging**